### PR TITLE
HDDS-7722. FSO buckets fail to invalidate open file table cache when committing a key

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?

Fix incorrect use of openKeyTable/openFileTable, where key/file name, instead of open key/file name is used.

## What is the link to the Apache JIRA

[HDDS-7722](https://issues.apache.org/jira/browse/HDDS-7722)

## How was this patch tested?

Existing unit tests

### CODE REVIEW

**Comment by kerneltime** [comment]
@duongkame can you please take a look?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequestWithFSO.java [hunk 1]
```diff
@@ -193,7 +193,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       }
 
       // Add to cache of open key table and key table.
-      OMFileRequest.addOpenFileTableCacheEntry(omMetadataManager, dbFileKey,
+      OMFileRequest.addOpenFileTableCacheEntry(omMetadataManager, dbOpenFileKey,
```
Comment (duongkame) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequestWithFSO.java:196]: Nice catch.

So, what happens is when an FSO key is committed, the open-key entry is not cleaned up. Logically, that should prevent the key from being reopened, e.g. for a overwrite. 
Fortunately (or maybe not), when OM opens a key for writing, it just doesn't care if the file has been opened already (even by the same client). I feel like this behavior results in some funny scenarios when multiple clients open and commit the same file. Guess it's something we can think about.

**Review by duongkame — APPROVED** [review]
LGTM. Good catch.

I have a concern regarding how OM handles multiple opens of the same file, but it's outside the scope of this fix.

**Review by errose28 — APPROVED** [review]
Thanks for the fix @jojochuang. LGTM.

For the record I think the bug is fairly benign.
- The operation leaves the cache is unchanged, since the request is to remove an invalid entry.
- The response class is making the correct changes to the DB
- Open key cleanup does not read the cache, so it will not pre-maturely clean up the key if it is not removed.
    - This prevents this from causing data loss
- Does not affect other clients since open keys are independent based on client ID.
- Does not affect the same client since a new request by the client to write to the same file would overwrite the cache entry that was not cleared.

**Comment by jojochuang** [comment]
Thanks Ethan. Good to know that you find it benign.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequestWithFSO.java (hunk 1 above)
Comment (errose28) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequestWithFSO.java:196]: @duongkame did [this](https://github.com/apache/ozone/pull/4156#pullrequestreview-1242797126) answer your question or do you still have some concerns about this change?

**Review by kaijchen — APPROVED** [review]
LGTM, thanks @jojochuang for the fix.

**Comment by kaijchen** [comment]
Thanks @duongkame @errose28 for the review.

**Pull request merged**